presto-main/src/test/java/com/facebook/presto/operator/TestHashJoinOperator.java

```diff
@@ -98,6 +98,7 @@
 import static java.util.concurrent.Executors.newScheduledThreadPool;
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static java.util.stream.Collectors.toList;
+import static org.testng.Assert.assertFalse;
```

Comment:
I didn't review this yet. But please add also tests to ATQ, but please make sure that right side cannot be recognized as empty during planning phase (you can use non-deterministic condition, like involving `rand()` that is deterministically false, like `rand() < 0`)

Reply:
Will do in the next iteration.

Reply:
@findepi I've updated the code to include all the suggested changes, plus I've added a few tests to ATQ to cover the scenarios touched by this code change.